sstmac/replacements/string.h

```diff
@@ -26,12 +26,26 @@
 extern "C" {
 #endif
 
+
+#if defined(__clang__) // Remove pragma warnings if clang
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunknown-pragmas"
```

Comment:
I think this is the correct way to do it. This header should only really get included when doing source-to-source. This is a good way to avoid warnings with GCC.

Reply:
The other option is to make a custom S2S define and only enable this code in that circumstance.